apps/prairielearn/src/batched-migrations/20250318180548_course_instance_usages__workspaces__backfill_v3.ts

```diff
@@ -0,0 +1,32 @@
+import { z } from 'zod';
+
+import { makeBatchedMigration } from '@prairielearn/migrations';
+import { loadSqlEquiv, queryAsync, queryRow } from '@prairielearn/postgres';
+
+const sql = loadSqlEquiv(import.meta.url);
+
+const END_DATE = '2025-03-21T00:00:00Z';
+
+export default makeBatchedMigration({
+  async getParameters() {
+    // First delete old usage data
+    await queryAsync(sql.delete_old_usages, { END_DATE });
+
+    // Only backfill from workspaces within the date range
+    const min = 1n;
+    const max = await queryRow(
+      sql.select_max_bound,
+      { END_DATE },
+      z.bigint({ coerce: true }).nullable(),
+    );
+    return { min, max, batchSize: 10_000 };
+  },
+  async execute(start: bigint, end: bigint): Promise<void> {
+    await queryAsync(sql.update_course_instance_usages_for_workspaces, {
+      start,
+      end,
+
```

Comment:
Micro-nit:
```suggestion
```

Reply:
Ha, I'm too used to prettier doing everything for me 😅. Fixed in e1debeffd71cfe2bd80421c8f702209cedc6a4e3